### PR TITLE
refactor: display tokens

### PR DESCRIPTION
**Pull request by agualis**
Main changes: 

- Removes all API deprecated `pool.displayTokens` calls.
- Simplifies token related components types by using `ApiToken` instead of types like `GqlDisplayToken` 
- Centralises token display helpers in a new file (`pool.tokens.utils.ts`). 
- Sorts display pool tokens by index to have a more predictable token display outputs. 
- Adds integration tests to cover all the new extracted functions above (covering all the pool edge cases).
- Introduces a new `PoolCore` type to start cleaning the mess around different Pool types (`Pool, PoolListItem, PoolData`...) used across the Pool token display related components. (This must be simplified even more in incoming PRs but we need it now to enable previous points).

`TODO`: 
`VoteListTableRow` and other components  in https://github.com/balancer/frontend-monorepo/pull/148 once this PR is merged into main. 
This PR contained the necessary code: https://github.com/balancer/frontend-monorepo/pull/373

### CODE REVIEW

File: packages/lib/modules/pool/pool.helpers.ts [hunk 1]
```diff
@@ -398,15 +398,21 @@ export function getVaultConfig(pool: Pool) {
...
-export function isV2Pool(pool: Pool): boolean {
+export function isV2Pool(
+  pool: Pick<Pool | PoolListItem | GqlPoolBase, 'protocolVersion'>
+): boolean {
   return pool.protocolVersion === 2
 }
 
-export function isV3Pool(pool: Pool | PoolListItem | GqlPoolBase): boolean {
+export function isV3Pool(
+  pool: Pick<Pool | PoolListItem | GqlPoolBase, 'protocolVersion'>
```
Comment (groninge01) [inline packages/lib/modules/pool/pool.helpers.ts:414]: 3rd instance -> move to a type?

File: packages/lib/modules/pool/pool.tokens.utils.ts
Comment (garethfuller) [inline packages/lib/modules/pool/pool.tokens.utils.ts:1]: Rename pool-tokens.utils.ts. The addition to the file extension should always be
a kind of file classification only. e.g. utils/helpers/types/etc.

Comment (agualis) [reply packages/lib/modules/pool/pool.tokens.utils.ts:1]: Done

File: apps/beets-frontend-v3/app/(app)/portfolio/[chain]/page.tsx [hunk 2]
```diff
@@ -73,7 +75,11 @@ export default function NetworkClaim() {
                         </Text>
                       </HStack>
                       <HStack w="full">
-                        <TokenIconStack chain={pool.chain} size={36} tokens={pool.displayTokens} />
+                        <TokenIconStack
+                          chain={pool.chain}
+                          size={36}
+                          tokens={getUserReferenceTokens(pool as PoolCore)}
```
[the remark this answers is not in the document]
Comment (agualis) [reply apps/beets-frontend-v3/app/(app)/portfolio/[chain]/page.tsx]: It was in my TODO list. I changed the PoolCore type to get rid of all the `as PoolCore` castings.

File: packages/lib/modules/pool/pool.helpers.ts (hunk 1 above)
Comment (agualis) [reply packages/lib/modules/pool/pool.helpers.ts]: Done 👍

**Review by garethfuller — APPROVED** [review]
